pkg/apis/serving/v1beta1/configmap.go

```diff
@@ -114,6 +116,19 @@ func NewIngressConfig(cli client.Client) (*IngressConfig, error) {
 		if ingressConfig.IngressGateway == "" || ingressConfig.IngressServiceName == "" {
 			return nil, fmt.Errorf("invalid ingress config - ingressGateway and ingressService are required")
 		}
+		if ingressConfig.PathTemplate != "" {
+			// TODO: ensure that the generated path is valid, that is:
+			// * both Name and Namespace are used to avoid collisions
+			// * starts with a /
+			// For now simply check that this is a valid template.
+			_, err := template.New("path-template").Parse(ingressConfig.PathTemplate)
+			if err != nil {
+				return nil, fmt.Errorf("Invalid ingress config, unable to parse pathTemplate: %v", err)
+			}
+			if ingressConfig.IngressDomain == "" {
+				return nil, fmt.Errorf("Invalid ingress config - igressDomain is required if pathTemplate is given")
+			}
```

Comment:
The `inferenceservice-config`  has the default value `example.com` so this check does not take effect , so let's add a comment in the configmap that `IngressDomain` is used in case of `RawDeployment ` and path based routing for `Serverless` mode. https://github.com/kserve/kserve/blob/b15cd9e43771aa90ae8f2e0094c4333197af6bd1/config/configmap/inferenceservice.yaml#L62

Reply:
@yuzisun isn't the default value assigned a few lines below if this is not given?

https://github.com/kserve/kserve/blob/c587ef1769f33f6134ba16b6ef1e7f569eb3eda4/pkg/apis/serving/v1beta1/configmap.go#L138